pkg/email/email.go

```diff
@@ -0,0 +1,72 @@
+package email
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"net/url"
+	"path"
+)
+
+// Config holds EASi application specific configs for SES
+type Config struct {
+	GRTEmail          string
+	URLHost           string
+	URLScheme         string
+	TemplateDirectory string
+}
+
+type templateCaller interface {
+	Execute(wr io.Writer, data interface{}) error
+}
+
+type templates struct {
+	systemIntakeSubmissionTemplate templateCaller
+}
+
+type sender interface {
+	Send(toAddress string, subject string, body string) error
+}
+
+// Client is an EASi SES client wrapper
+type Client struct {
+	config    Config
+	templates templates
+	sender    sender
+}
+
+func templateError(name string) error {
+	return fmt.Errorf("failed to get template: %s", name)
+}
+
+// NewClient returns a new email client for EASi
+func NewClient(config Config, sender sender) (Client, error) {
+	rawTemplates, err := template.ParseGlob(path.Join(config.TemplateDirectory, "*.gohtml"))
+	if err != nil {
+		return Client{}, err
+	}
+	appTemplates := templates{}
+
+	systemIntakeSubmissionTemplateName := "system_intake_submission.gohtml"
```

Comment:
might change this, but essentially checking that the required files exist on instantiation rather than send